From adb69df26914bf926b43fa6e6894800e014dd966 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 12 Dec 2021 01:01:01 +0000
Subject: [PATCH 6/6] gnu: Add hcl v2

* gnu/packages/golang.scm: (go-github-com-hashicorp-hcl-v2): New variable.
---
 gnu/packages/golang.scm | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 50f1947dba..699bc49401 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -2360,6 +2360,42 @@ (define-public go-github-com-hashicorp-hcl
       (home-page "https://github.com/hashicorp/hcl")
       (license license:mpl2.0))))
 
+(define-public go-github-com-hashicorp-hcl-v2
+  (package
+    (name "go-github-com-hashicorp-hcl-v2")
+    (version "2.11.1")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/hashicorp/hcl")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32
+         "0f9flmmkj7fr1337fc56cqy73faq87ix375hnz3id4wc023przv1"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "github.com/hashicorp/hcl/v2"))
+    (native-inputs
+     `(("go-github-com-davecgh-go-spew" ,go-github-com-davecgh-go-spew)))
+    (inputs
+     `(("go-github-com-agext-levenshtein"
+        ,go-github-com-agext-levenshtein)
+       ("go-github-com-mitchellh-go-wordwrap"
+        ,go-github-com-mitchellh-go-wordwrap)
+       ("go-github-com-zclconf-go-cty"
+        ,go-github-com-zclconf-go-cty)
+       ("go-github-com-apparentlymart-go-textseg-v13"
+        ,go-github-com-apparentlymart-go-textseg-v13)))
+    (synopsis "Go implementation of HashiCorp Configuration Language V2")
+    (description
+     "This package contains the main implementation of the @acronym{HCL,
+HashiCorp Configuration Language}.  HCL is designed to be a language for
+expressing configuration which is easy for both humans and machines to read.")
+    (home-page "https://github.com/hashicorp/hcl")
+    (license license:mpl2.0)))
+
 (define-public go-golang-org-x-tools
   (let ((commit "8b927904ee0dec805c89aaf9172f4459296ed6e8")
         (revision "0"))
-- 
2.34.0

